DependencyInjection/Configuration.php

```diff
@@ -185,136 +186,280 @@ private function configureClients(ArrayNodeDefinition $root)
+                    ->end()
+                ->end()
+                ->arrayNode('header_defaults')
+                    ->canBeEnabled()
+                    ->info('Set header to default value if it does not exist.')
+                    ->fixXmlConfig('header')
+                    ->children()
+                        ->arrayNode('headers')
+                            ->info('Keys are the header names, values the header values')
+                            ->normalizeKeys(false)
+                            ->useAttributeAsKey('name')
+                            ->prototype('scalar')->end()
                         ->end()
-                    ->end() // End decoder plugin
-
-                    ->arrayNode('history')
+                    ->end()
+                ->end()
+                ->arrayNode('header_set')
                     ->canBeEnabled()
-                        ->children()
-                            ->scalarNode('journal')
-                                ->info('This must be a service id to a service implementing Http\Client\Plugin\Journal')
-                                ->isRequired()
-                                ->cannotBeEmpty()
-                            ->end()
+                    ->info('Set headers to requests. If the header does not exist it wil be set, if the header already exists it will be replaced.')
+                    ->fixXmlConfig('header')
+                    ->children()
+                        ->arrayNode('headers')
+                            ->info('Keys are the header names, values the header values')
+                            ->normalizeKeys(false)
+                            ->useAttributeAsKey('name')
+                            ->prototype('scalar')->end()
                         ->end()
-                    ->end() // End history plugin
-
-                    ->arrayNode('logger')
-                    ->canBeDisabled()
-                    ->addDefaultsIfNotSet()
-                        ->children()
-                            ->scalarNode('logger')
-                                ->info('This must be a service id to a service implementing Psr\Log\LoggerInterface')
-                                ->defaultValue('logger')
-                                ->cannotBeEmpty()
-                            ->end()
-                            ->scalarNode('formatter')
-                                ->info('This must be a service id to a service implementing Http\Message\Formatter')
-                                ->defaultNull()
-                            ->end()
+                    ->end()
+                ->end()
+                ->arrayNode('header_remove')
+                    ->canBeEnabled()
+                    ->info('Remove headers from requests.')
+                    ->fixXmlConfig('header')
+                    ->children()
+                        ->arrayNode('headers')
+                            ->info('List of header names to remove')
```

Comment:
because for the others, the header names are the keys, while here the headers are the values. only the keys are normalized, never the values.